cli/src/config.rs

```diff
@@ -324,7 +324,28 @@ fn env_base() -> config::Config {
         builder = builder.set_override("ui.color", "never").unwrap();
     }
     if let Ok(value) = env::var("PAGER") {
-        builder = builder.set_override("ui.pager", value).unwrap();
+        builder = builder
+            .set_override(
+                "ui.pager.command",
+                config::Value::new(
+                    None,
+                    config::ValueKind::Array(vec![config::Value::new(
+                        None,
+                        config::ValueKind::String(value),
+                    )]),
+                ),
```

Comment:
This is pretty ugly. Is there a more convenient API in the `config` crate?

Reply:
`.set_override("ui.pager.command", vec![value])` appears to work.
https://docs.rs/config/0.13.4/config/enum.ValueKind.html#impl-From%3CVec%3CT%3E%3E-for-ValueKind

Reply:
Actually, I'm not sure this is correct if PAGER contains a command with arguments, e.g. `PAGER="less -FRX"`?

Reply:
It's possible this is still worth doing even if it doesn't work for `PAGER="less -FRX"`, I'm unsure how common that is.

Or, we could allow `ui.pager.command` to be a string.

Reply:
> Actually, I'm not sure this is correct if PAGER contains a command with arguments, e.g. `PAGER="less -FRX"`?

Good point. `PAGER` needs to be parsed as if it were executed by the shell. It might be simpler to just special case `PAGER=less`.

Reply:
I'll wait a bit to hear from others if we actually want this PR before I spend more time on it.

Reply:
Judging by https://github.com/martinvonz/jj/issues/4745, this is in fact an issue.

The easiest solution that comes to my mind would be to set `LESS=FRX` if `PAGER=less` and `LESS` is unset, pretty much as Yuya suggested. If `ui.pager="less"`, we could also set LESS.

Does that sound all right? It's a slightly ugly special case, but I don't see a much better solution. (Actually, if we wanted to support many such special cases, we could have a config `[pager.less]` analogous to `merge-tools.less`, and consider `less` a key into it, but this could be confusing and is probably not worth it at this point).

Should I make a PR or do we want to use this one?

Reply:
The other approach that makes some sense to me is to always forcibly add `R` to LESS, whatever its value is, or just setting `LESS=R` if it's not defined. But I think this is special-casing `jj` to `less`'s behavior even more.

Reply:
Maybe we can ignore `$PAGER == "less"` or `$PAGER` at all? It's the default on non-Windows, and it's less likely that Windows users customize `$PAGER` based on their preference.

Reply:
I'd like to respect `PAGER="less -RX"`. Missing the `F` is important if, say, you are calling `jj` from `ranger` or Midnight Commander/.

Reply:
For most people, I'd probably recommend using `LESS=RX` instead of `PAGER="less -RX"`, but I know I did the latter for a while. Not everybody knows about the `LESS` environment variable.

Reply:
One option is to set `LESS=R` (instead of FRX) if LESS is unset, without any further conditions. This seems safe to me, but a worse experience for people with bad defaults. (As far as I'm concerned, LESS=R should be the default compiled into `less`).

I think I prefer the approach I suggested above, but I'm not sure, both have pros and cons.

Reply:
Ok, then just ignore `$PAGER == "less"`? fwiw, `$PAGER=whatever` works only if you didn't set `ui.pager` config.

Reply:
That works on Unix, but I'm not sure that makes sense on Windows where we default to the builtin pager.

OTOH, even on Windows, if `PAGER=less`, we want to add `LESS=R` or `FRX`.

One option would be to ignore PAGER=less, but print a warning when we do so on Windows (probably no need on Unix).

Reply:
Printing a warning might also make sense since, AFAIK, some Windows versions of less are broken. I don't have first-hand experience with that, but I seem to remember some discussions that only Git Bash less really works.

Reply:
Alternatively, warn (on any platform) if `$PAGER == "less"`, and if `$LESS=R` is not set, and if `ui.pager` is not set, and falls back to the jj's default.

Rationale: the default set by system bashrc is often incompatible with `jj`. User should fix the problem.

Reply:
We could start with this. Do you know, if we suggest the following command, will it work on Windows?

    jj config set --user ui.pager "less -FRX"

Reply:
Thinking it a bit more, adding a warning would be complicated than the other options.

In order of ease of implementation/documentation:
1. ignore `$PAGER`
2. ignore `$PAGER == "less"`
3. set `$LESS` if `$PAGER == "less"` and if `$LESS` is unset (which is hg and git do iirc)
4. warn if `$PAGER == "less"`, not `$LESS`, not `ui.pager`

Reply:
According to a [Discord discussion](https://github.com/NixOS/nixpkgs/blob/c6f1ef99d134c51556cf01db04d33074ebbbf43c/nixos/modules/programs/less.nix#L21), this could create a false alarm on Nix that sets `-R` using a lesskey file.

Perhaps suggesting `jj config set --user ui.pager "less -RX"` or `LESS=RX` to them is not too much trouble, I'm not sure.

Reply:
Good idea to compare to Git. I found

https://github.com/git/git/blob/8f8d6eee531b3fa1a8ef14f169b0cb5035f7a772/Makefile#L357

Not sure what LV is about, but just following what they do seems OK to me.

Reply:
I think git sets it regardless of PAGER, anytime LESS is unset.

Reply:
But I somehow can't reproduce this. I unset `LESS`, `core.pager` seems unset, but `PAGER="less -R" git diff` acts differently from `PAGER="less -R +F" git diff`. OTOH, `PAGER="less +R" git diff` prints color. So, I don't know what's going on with Git.

Reply:
Re @yuja 's options from https://github.com/martinvonz/jj/pull/3657/files#r1833607331, I think setting `LESS` whenever it is unset would be between 2 and 3.

Reply:
> Re @yuja 's options from https://github.com/martinvonz/jj/pull/3657/files#r1833607331, I think setting `LESS` whenever it is unset would be between 2 and 3.

yes, it would be relatively easy because we just need to update `Ui::new_paged()`.

In my list, 1-3 would be handled in config layer. 4 needs more context.